src/components/autocomplete-item.tsx

```diff
@@ -40,9 +40,14 @@ const AutocompleteItem = ({
         onClick={(e) => handleOnClick(item, e)}
         className={active ? 'hover' : ''}
       >
-        {substringToHighlight
-          ? highlightSubstring(item.pathLabel, substringToHighlight)
-          : item.itemLabel}
+        {substringToHighlight ? (
+          <SubstringHighlight
+            string={item.pathLabel}
+            substring={substringToHighlight}
+          />
```

Comment:
I'd prefer a component API similar to other components where we modify text before rendering it (e.g. LongNumber) where the text is a child of the component.
```suggestion
          <SubstringHighlight substring={substringToHighlight}>
            {item.pathLabel}
          </SubstringHighlight>
```
We can enforce the type of children to be only a string in this case

Reply:
Good point